ruby/ql/lib/codeql/ruby/frameworks/ActiveRecord.qll

```diff
@@ -176,11 +176,16 @@ private predicate sqlFragmentArgumentInner(DataFlow::CallNode call, DataFlow::No
     activeRecordQueryBuilderCall([
         "delete_all", "delete_by", "destroy_all", "destroy_by", "exists?", "find_by", "find_by!",
         "find_or_create_by", "find_or_create_by!", "find_or_initialize_by", "find_by_sql", "from",
-        "group", "having", "joins", "lock", "not", "order", "reorder", "pluck", "where", "rewhere",
-        "select", "reselect"
+        "having", "lock", "not", "where", "rewhere"
       ]) and
   sink = call.getArgument(0)
   or
+  call =
+    activeRecordQueryBuilderCall([
+        "group", "joins", "order", "reorder", "pluck", "select", "reselect"
```

Comment:
We should probably also move [`where`](https://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-where) here as it can accept multiple arguments.

Reply:
It looks like the additional arguments of `where` are not always vulnerable to sql sinks; as when the first argument is a template with `?`s then subsequent arguments are used safely as template parameters.

Reply:
Ah, yes this is fine as is then.